pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -48,39 +49,61 @@ func (d *DualWriterMode2) Mode() DualWriterMode {
 }
 
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
-func (d *DualWriterMode2) Create(ctx context.Context, original runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+func (d *DualWriterMode2) Create(ctx context.Context, in runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	var method = "create"
 	log := d.Log.WithValues("method", method)
 	ctx = klog.NewContext(ctx, log)
 
+	accIn, err := meta.Accessor(in)
+	if err != nil {
+		return nil, err
+	}
+	if accIn.GetUID() != "" {
+		return nil, fmt.Errorf("UID should not be present: %v", accIn.GetUID())
+	}
+
+	if accIn.GetName() == "" && accIn.GetGenerateName() == "" {
```

Comment:
I think we could actually do this:
```
	if accIn.GetGenerateName() != "" {
		return nil, fmt.Errorf("generateName cannot be set")
	}

	if accIn.GetName() == "" {
accIn.SetName(names.SimpleNameGenerator.GenerateName(accIn.GetGenerateName()))
	}
	```

Reply:
i believe this would add behavior that doesn't exist in etcd compatible mode 5, so behavior would change on clients as we progress through the modes

Reply:
for that reason I think it's better to add this on every mode, no? I think it should not be possible for generatename to be present when creating an object. And since the name has to match in both stores, we need to add it on the dual writer part.

Reply:
> better to add this on every mode

my understanding is that the dual writer will only exist during a migration period, not long term, so it seems like mode 5 should work without modifications. is that correct?

> should not be possible for generatename to be present when creating an object

at which level? at the client request level, either name or generateName need to be present in the request, but not both at the same time

Reply:
> my understanding is that the dual writer will only exist during a migration period, not long term, so it seems like mode 5 should work without modifications. is that correct?

Correct... but even in mode 5, or like, without any dualwriter, should we allow for both to be empty? If not, I added a condition [here](https://github.com/grafana/grafana/pull/93852/files#diff-a6d40da7a17d43622287c2147606ef055a394f2110ac283dea3c10d5fe64f216R28-R37). At least, this was what I understood from the naming issue, so that we can pass the name down on objInfo object and we are actually able to modify the object we want instead of creating a new entry on the DB (which was happening before). Please corect me if I'm wrong here!

>at which level? at the client request level, either name or generateName need to be present in the request, but not both at the same time
This condition was added on the client level

Reply:
I am also unsure about enforcing this in the dual writer. @leonorfmartins could you walk me through the generated name issue ? I this also solved by the fact we are not writing to unistore the response from legacy ?

Reply:
so, are empty names something we will want to avoid anyways, even in mode5? I don't think we will need to enforce this here anymore, but I'm wondering if we should add this fail check on the server side.

Reply:
In any case, I removed it now as the name issue got solved with reusing the returning object from legacy to write to UN.